crates/iceberg/Cargo.toml

```diff
@@ -41,20 +41,24 @@ either = "1"
 futures = "0.3"
 itertools = "0.11"
 lazy_static = "1"
+log = "^0.4"
 murmur3 = "0.5.2"
 once_cell = "1"
 opendal = "0.40"
 ordered-float = "4.0.0"
+reqwest = { version = "^0.11", features = ["json"] }
```

Comment:
Do we need to put rest catalog in a new crate? I'm guessing not all iceberg users need this.

Reply:
Maybe we should make it a feature? I'm not sure if it deserves another crate. cc @JanKaul  @ZENOTME How do you guys think?

Reply:
The problem with the features is that they are add-only, making it difficult for users to disable them.

Reply:
Sorry, I don't get your point, would you give a concrete example? 
 One concern with separate crate approach is that it makes loading catalog dynamically like Python difficult.

Reply:
I'm also in favor of using a separate crate for the rest catalog. This way we avoid bringing in dependencies for catalogs that don't need them and that might prevent the crate from compiling, like wasm/wasi for example.

Reply:
Cool, I'll move it a separate crate.